Modules/Core/Common/include/itkImageBase.h

```diff
@@ -242,6 +242,14 @@ class ITK_TEMPLATE_EXPORT ImageBase : public DataObject
   virtual void
   Allocate(bool initialize = false);
 
+  /** Allocates the pixel buffer of the image, zero-initializing its pixels. `AllocateZeroInitializedPixelBuffer()` is
+   * equivalent to `Allocate(true)`. It is just intended to make the code more readable. */
+  void
+  AllocateZeroInitializedPixelBuffer()
+  {
+    return this->Allocate(true);
```

Comment:
_Allocate_ in _ImageBase_ is just an empty function, BTW.

Reply:
Indeed, `ImageBase::Allocate(bool)` is empty. Ideally, it should have been a pure virtual function, but apparently declaring `Allocate(bool) = 0` would break backwards compatibility:

https://github.com/InsightSoftwareConsortium/ITK/blob/2c264ea2ef62e916c6ef947599ce659cc8fce5fe/Modules/Core/Common/include/itkImageBase.h#L235-L243

Maybe we should mark the (empty) function body of `ImageBase::Allocate(bool)`  `ITK_FUTURE_LEGACY_REMOVE`. Something to consider for ITK 6... 🤔 But that's beyond the scope of this PR anyway  🤷

Reply:
Obviously declaring `ImageBase::Allocate` _pure virtual_ would make `ImageBase` an abstract base class. So it would disallow creating `ImageBase` objects. Probably `ImageBase::New()` should then also be removed.

But currently it is still possible to create `ImageBase` objects. Such an object would just contain the image grid/size/geometry information, _without_ the pixel data. I can imagine that that _might_ be a useful feature. Actually I'm interested to hear if people actually use that feature! 😃

Reply:
Thanks @blowekamp I did not yet realize that ITKv4 registration would use images that way!

I think this resolves the conversation, but the pull request is still open  😃 Please approve, if you agree about adding an `ImageBase::AllocateZeroInitializedPixelBuffer()` member function!